### ISSUE TITLE
Source of shard in shard.lock

### ISSUE THREAD

**Issue by jwoertink** [crystal-lang/shards#268]
I have a lot of shards in my yml file, but when I boot my app, it fails on a shard that I never included. When I look in the shard.lock file, I see the shard that fails, but I don't know where it came from. It's obviously included as a dependency of a shard that I did include, but having that information available would be super helpful. Ruby gems does this with the Gemfile.lock 

The activejob gem requires the activesupport and globalid gems as dependencies. 
```
activejob (5.1.4)
    activesupport (= 5.1.4)
    globalid (>= 0.3.6)
```

Showing all the children gets a bit cumbersome in large projects, so maybe we could do something like this where we just show the parent:

```
avram:
  github: luckyframework/avram
  version: 0.9.4
  from: lucky

lucky:
  github: luckyframework/lucky
  version: 0.14.0
```

This would denote that I didn't actually include avram in my shard.yml file, but it was included from lucky which I did include in mine.

**Comment by ysbaddaden** [comment]
`shards list --tree` alloss that. Pipe it to less, for example, search for the dependency, and you'll know how it got included (even if nested).

**Comment by jwoertink** [comment]
hmm... I'm not sure. Is this what you're talking about?

```
[11:19AM] typein (develop)$ shards list --tree
Shards installed:
  * lucky (0.14.0)
  * lucky_cli (0.14.0)
  * teeplate (0.7.0)
  * habitat (0.4.3)
  * wordsmith (0.2.0)
  * avram (0.9.4)
  * blank (0.1.0)
  * lucky_cli (0.14.0)
  * teeplate (0.7.0)
  * db (0.5.1)
```

From this list, I can't tell which shard requires `blank` as a dependency. This seems to only list all of the shards just in a more concise way. Maybe this `--tree` flag broke at some point? It seems to print the exact same as just doing `shards list`

```
[11:24AM] typein (develop)$ diff <(shards list) <(shards list --tree)
[11:25AM] typein (develop)$
```

**Comment by ysbaddaden** [comment]
Oh, I thought it got released a while back, but it's actually only available since 0.9.0.beta. Can you try the latest prerelease?

**Comment by jwoertink** [comment]
ah! Ok, so that version does have it. I think this is good enough for my use case then. I'll use the beta locally until it's released. Thanks!

**Comment by straight-shoota** [comment]
Yeah, `shards list --tree` should work for this. But I think we should have a `shards why <shard>` command at some point as well. It would directly show the reverse dependency tree for `<shard>`.

**Comment by ysbaddaden** [comment]
No. List tree solves that issue and more already. No need to be more specific.

**Comment by straight-shoota** [comment]
But it's hard to parse, especially when a dependency appears in different branches and deeper in the tree.

Providing a user-friendly feature would be helpful. I don't say it needs to be added right now.
